frontend/ui/elements/screen_rotation_menu_table.lua

```diff
@@ -0,0 +1,124 @@
+local _ = require("gettext")
+local Device = require("device")
+local Event = require("ui/event")
+local FileManager = require("apps/filemanager/filemanager")
+local UIManager = require("ui/uimanager")
+local Screen = Device.screen
+local S = require("ui/data/strings")
+
+return {
+    text = _("Rotation"),
+    sub_item_table_func = function()
+        local rotation_table = {}
+
+        if Device:canToggleGSensor() then
+            table.insert(rotation_table, {
+                text = _("Ignore accelerometer rotation events"),
+                checked_func = function()
+                    return G_reader_settings:isTrue("input_ignore_gsensor")
+                end,
+                callback = function()
+                    G_reader_settings:flipNilOrFalse("input_ignore_gsensor")
+                    Device:toggleGSensor(not G_reader_settings:isTrue("input_ignore_gsensor"))
+                end,
+            })
+        end
+
+        table.insert(rotation_table, {
+            text = _("Keep file browser rotation"),
+            help_text = _("When checked the rotation of the file browser and the reader will not affect each other"),
+            checked_func = function()
+                return G_reader_settings:isTrue("lock_rotation")
+            end,
+            callback = function()
+                G_reader_settings:flipNilOrFalse("lock_rotation")
+            end,
+            separator = true,
+        })
+
+        if FileManager.instance then
+            table.insert(rotation_table, {
+                text_func = function()
+                    local text = S.LANDSCAPE_ROTATED
+                    if G_reader_settings:readSetting("fm_rotation_mode") == 3 then
```

Comment:
Could probably use the fancy constants from https://github.com/koreader/koreader-base/blob/68b3a7c5d74cb77d82a9eff5b9d9dd9c5bb368f7/ffi/framebuffer.lua#L51-L55 throughout, instead of magic numbers ;).